mucommander-core/src/main/java/com/mucommander/ui/text/SelectAllOnFocusListener.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * This file is part of muCommander, http://www.mucommander.com
+ *
+ * muCommander is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * muCommander is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+package com.mucommander.ui.text;
+
+import java.awt.event.FocusEvent;
+import java.awt.event.FocusListener;
+import javax.swing.text.JTextComponent;
+
+
+/**
+ * @author Gerolf Scherr
+ */
+public class SelectAllOnFocusListener implements FocusListener {
+
+    private final static class InstanceHolder {
+        final static SelectAllOnFocusListener me = new SelectAllOnFocusListener();
+    }
```

Comment:
what the benefit in having the `InstanceHolder` class? the static instance can rather be a member of the `SelectAllOnFocusListener` class

Reply:
this pattern avoids synchronization issues. the inner class will be initialized via the class loader, which runs synchronized. see:
https://en.wikipedia.org/wiki/Initialization-on-demand_holder_idiom

Reply:
I see, I don't think the lazy evaluation is important in this case but if we try to minimize the memory consumption, it's also ok to create a new instance of the listener per-field - the search dialog is rarely used so objects would be allocated and garbage collected rarely in this context